docs/migration/migrating-enact-apps.md

```diff
@@ -0,0 +1,531 @@
+</IncrementSlider>
+```
+The `onDecrement` and `onIncrement` props have been removed.  Developers can use `onKeyDown` and/or `onKeyUp`.
+
+The `detachedKnob` and `scrubbing` props have been removed with no replacement.
+
+#### `Input`
+The `<input>` element's styles have changed for `height`, `vertical-align`, and `margin`s.  Some previously used
+sizing and positioning CSS may no longer be necessary.  Developers should verify their layouts.
+
+#### `Item.OverlayDecorator`, `Item.Overlay`, and `Item.ItemOverlay`
+These components are all replaced by `moonstone/SlotItem`.
+
+#### `Marquee.MarqueeText`
+This component is replaced by `moonstone/Marquee`.
+
+#### `Popup`
+The `containerId` prop has changed to `spotlightId`.
+##### Example
+###### 1.x
+```
+<Popup containerId="spotlightContainer" />
+```
+###### 2.0
+```
+<Popup spotlightId="spotlightContainer" />
+```
+
+#### `ProgressBar`
+The boolean props `horizontal` and `vertical` have been replaced by the `orientation` prop.
+##### Example
+###### 1.x
+```
+<ProgressBar /> // `horizontal` prop is `true` by default.
+<ProgressBar vertical />
+```
+###### 2.0
+```
+<ProgressBar /> // `orientation` prop is `"horizontal"` by default.
+<ProgressBar orientation="vertical" />
+```
+
+#### `Scroller`
+The boolean props `horizontal` and `vertical` have been replaced by the `direction` prop.
+##### Example
+###### 1.x
+```
+<Scroller /> // `horizontal` prop is `true` by default.
+<Scroller vertical />
+```
+###### 2.0
+```
+<Scroller /> // `direction` prop is `"horizontal"` by default.
+<Scroller direction="vertical" />
+```
+The `scrollTo` method's `indexToFocus` option has been removed.  Use the `focus` option and
+scroll by `index` (only applicable for components like `VirtualList`) or `node`.
+##### Example
+###### 1.x
+```
+...
+const cbScrollTo = () => {
+	this.scrollTo({indexToFocus: 1});
+};
+...
+<VirtualList cbScrollTo={cbScrollTo} ... />
+...
+```
+###### 2.0
+```
+...
+const cbScrollTo = () => {
+	this.scrollTo({focus: true, index: 1});
+};
+...
+<VirtualList cbScrollTo={cbScrollTo} ... />
+...
+```
+
+#### `Slider`
+The exports for `SliderFactory` and `SliderBaseFactory` have been removed.  If your code
+imported these, you should use `ui/Slider` and `ui/SliderBase` instead.
+##### Example
+###### 1.x
+```
+import {SliderFactory} from '@enact/moonstone/Slider';
+import {SliderBaseFactory} from '@enact/moonstone/Slider';
+```
+###### 2.0
+```
+import Slider from '@enact/ui/Slider';
+import {SliderBase} from '@enact/ui/Slider';
+```
+The boolean props `horizontal` and `vertical` have been replaced by the `orientation` prop.
+##### Example
+###### 1.x
+```
+<Slider /> // `horizontal` prop is `true` by default.
+<Slider vertical />
+```
+###### 2.0
+```
+<Slider /> // `orientation` prop is `"horizontal"` by default.
+<Slider orientation="vertical" />
+```
+The `tooltipAsPercent`, `tooltipSide`, and `tooltipForceSide` props for the built-in tooltip
```

Comment:
`forceSide` was actually removed altogether and replaced with a generalized `side` prop that supports both locale-aware and locale-independent values.

1.x => `<Slider tooltip tooltipForceSide tooltipSide="after" orientation="vertical" />`
2.x => `<Slider orientation="vertical"><SliderTooltip side="right" /></Slider>`